### ISSUE TITLE
Incorrect Linebreaks

### ISSUE THREAD

**Issue by tpreusse** [orbiting/pdf#140]
[fixtures/linebreaks](http://localhost:3007/fixtures/linebreaks)

<img width="499" alt="screenshot 2019-02-21 at 11 03 15" src="https://user-images.githubusercontent.com/410211/53160528-6d993800-35c8-11e9-87a7-1a8e0b96ff1f.png">

Looks like off by one.

Paragraph AST:

```json
{
  "children": [
    {
      "type": "text",
      "value": "Angeklagt ist der PR-Mann Sacha Wigdorovits. Die Staats­anwaltschaft wirft ihm versuchte Nötigung beziehungsweise Anstiftung oder Gehilfen­schaft dazu vor. Ausserdem geht es um das Aufbewahren, Auswerten, Zugänglich­machen und Weitergeben von unbefugt aufgenommenen Gesprächen. Für Sacha Wigdorovits gilt die Unschuldsvermutung."
    }
  ],
  "type": "paragraph"
}
```

It does contain soft-hyphen, but the bug occurs without them too.

Prd example: [`/2019/02/19/jetzt-kommt-es-zum-prozess-in-der-causa-geri-mueller.pdf`](https://cdn.republik.space/pdf/2019/02/19/jetzt-kommt-es-zum-prozess-in-der-causa-geri-mueller.pdf) page 2, 2nd paragraph.

**Comment by tpreusse** [comment]
I also inspected the hyphenation callback out of curiosity:
```
export default (string) => {
  console.log('in ', string)
  const r = hyphenate(string)
  console.log('out', r)
  return r
}
```
<img width="189" alt="screenshot 2019-02-21 at 11 30 48" src="https://user-images.githubusercontent.com/410211/53162522-7f7cda00-35cc-11e9-9502-0c528e133283.png">
<img width="163" alt="screenshot 2019-02-21 at 11 31 16" src="https://user-images.githubusercontent.com/410211/53162523-7f7cda00-35cc-11e9-83ab-3b2e406b3041.png">
<img width="176" alt="screenshot 2019-02-21 at 11 31 45" src="https://user-images.githubusercontent.com/410211/53162525-7f7cda00-35cc-11e9-8de9-2551dd4ab6e4.png">

The pre-hyphenated words come in in parts, this is a bit unexpected, the default hyphenation callback and ours use logic to skip hyphenation if done manually:
```js
  if (string.includes(SOFT_HYPHEN)) {
    return string.split(SOFT_HYPHEN)
  }
```
However if those things come in in parts it won't take any effect. @diegomura should I open a separate issue for this?

Also at the end of our article a not rendered undefined goes through the hyphenation callback as a string:
<img width="483" alt="screenshot 2019-02-21 at 11 37 32" src="https://user-images.githubusercontent.com/410211/53162855-311c0b00-35cd-11e9-8246-1b9b90fdc78e.png">
<img width="724" alt="screenshot 2019-02-21 at 11 38 51" src="https://user-images.githubusercontent.com/410211/53162907-485af880-35cd-11e9-84aa-5c3a7ba4f455.png">

**Comment by diegomura** [comment]
Thanks for reporting this @tpreusse . It's weird indeed 🤔 
You can open an issue on the react-pdf repo, but we can use this one to track this issue. As you prefer. 
I will take a look asap and see what's going on!

**Comment by diegomura** [comment]
With new textkit version:

- `/2019/02/19/jetzt-kommt-es-zum-prozess-in-der-causa-geri-mueller.pdf `
<img width="664" alt="Screen Shot 2019-04-08 at 1 04 15 PM" src="https://user-images.githubusercontent.com/5600341/55739166-e597b200-59fe-11e9-8c2d-e05e0ccc19ab.png">

**Comment by tpreusse** [comment]
fixed with #144
